### ISSUE TITLE
uTorrentPostProcess not converting

### ISSUE THREAD

**Issue by thejacobsons** [mdhiggins/sickbeard_mp4_automator#316]
Any time I grab a file other than mp4 in uTorrent, I have to manually convert the file even though it should be automatically converting them. 

These are my settings in autoProcess.ini

[uTorrent]
convert = True
couchpotato-label = movies
sonarr-label = tv
bypass-label = bypass
webui = True
action_before = stop
action_after = removedata
host = http://localhost:9093/gui/
username = 
password = 

I am running OS X and uTorrent WebUI v0.388. A screen shot of my Run Program:

<img width="659" alt="utor" src="https://cloud.githubusercontent.com/assets/8327120/9653993/05b0943a-51ed-11e5-82b7-d417a607d541.png">

Is anyone else having this issue or any idea what might be the problem?

TIA!
Dianna

**Comment by mdhiggins** [comment]
Is there any indication in the info.log file that utorrent script is actually being executed.

**Comment by thejacobsons** [comment]
No indication that it is being executed. I tried changing the script path and removed gui/ from the ini file. Neither of those worked.

**Comment by mdhiggins** [comment]
I think the problem is that uTorrent isn't able to execute the script for whatever reason. You'll have to play around with your uTorrent settings as well as make sure that the .py file has appropriate permissions (execute permissions). I haven't done too much testing with uTorrent on OSX but it should work once you get the execution to actually take place.

I would also highly recommend deluge over uTorrent, but that's just a personal preference.

**Comment by thejacobsons** [comment]
Ok, I finally had time to sit down and check this out. I try to run the uTorrentPostProcess.py file manually and the following happens:

/Applications/media/mp4_automator/uTorrentPostProcess.py [21:09:22]  Executing: /Applications/media/mp4_automator/uTorrentPostProcess.py "tv" "udp://tracker.publicbt.com/announce" "/Users/me/Documents/tv/Life.in.Pieces.S01E03.HDTV.XviD-FUM[ettv]" "multi" "Life.in.Pieces.S01E03.HDTV.XviD-FUM[ettv].avi" "93B0BDFF98C9E7F7060C0B823D8FE7FB9F9B5226"
2015-10-07 14:08:27 - uTorrentPostProcess - INFO - uTorrent post processing started.
2015-10-07 14:08:27 - readSettings - WARNING - Audio bitrate was invalid, defaulting to 256 per channel.
2015-10-07 14:08:27 - readSettings - ERROR - Unable to set tag language, defaulting to English.
Traceback (most recent call last):
  File "/Applications/media/mp4_automator/readSettings.py", line 367, in __init__
    babel = Language.fromalpha3(self.taglanguage)
  File "/Applications/media/mp4_automator/babelfish/language.py", line 110, in fromcode
    return cls(*language_converters[converter].reverse(code))
  File "/Applications/media/mp4_automator/babelfish/converters/__init__.py", line 253, in __getitem__
    raise KeyError(name)
KeyError: 'alpha3'
2015-10-07 14:08:27 - uTorrentPostProcess - ERROR - No valid label detected.

Apparently no label is detected even though I have labels as shown in my initial post and labels setup in uTorrent. The alpha3 error also isn't the issue because I can successfully run manual.py on files (but still get the alpha3 error). I have tried changing the label and I still receive the same 'No valid label detected' error.

**Comment by thejacobsons** [comment]
Ok, I finally had time to sit down and check this out. I try to run the uTorrentPostProcess.py file manually and the following happens:

/Applications/media/mp4_automator/uTorrentPostProcess.py [21:09:22] Executing: /Applications/media/mp4_automator/uTorrentPostProcess.py "tv" "udp://tracker.publicbt.com/announce" "/Users/me/Documents/tv/Life.in.Pieces.S01E03.HDTV.XviD-FUM[ettv]" "multi" "Life.in.Pieces.S01E03.HDTV.XviD-FUM[ettv].avi" "93B0BDFF98C9E7F7060C0B823D8FE7FB9F9B5226"
2015-10-07 14:08:27 - uTorrentPostProcess - INFO - uTorrent post processing started.
2015-10-07 14:08:27 - readSettings - WARNING - Audio bitrate was invalid, defaulting to 256 per channel.
2015-10-07 14:08:27 - readSettings - ERROR - Unable to set tag language, defaulting to English.
Traceback (most recent call last):
File "/Applications/media/mp4_automator/readSettings.py", line 367, in init
babel = Language.fromalpha3(self.taglanguage)
File "/Applications/media/mp4_automator/babelfish/language.py", line 110, in fromcode
return cls(*language_converters[converter].reverse(code))
File "/Applications/media/mp4_automator/babelfish/converters/init.py", line 253, in getitem
raise KeyError(name)
KeyError: 'alpha3'
2015-10-07 14:08:27 - uTorrentPostProcess - ERROR - No valid label detected.

Apparently no label is detected even though I have labels as shown in my initial post and labels setup in uTorrent. The alpha3 error also isn't the issue because I can successfully run manual.py on files (but still get the alpha3 error). I have tried changing the label and I still receive the same 'No valid label detected' error.

**Comment by mdhiggins** [comment]
Only thing I can think of is that your label isn't in the right spot in your command line ordering, I would go back and double check that info and see if thats the problem

**Comment by thejacobsons** [comment]
It is, the label is tv and is in the first variable slot as required by the script.

uTorrentPostProcess.py "**tv**" "udp://tracker.publicbt.com/announce" "/Users/me/Documents/tv/Life.in.Pieces.S01E03.HDTV.XviD-FUM[ettv]" "multi" "Life.in.Pieces.S01E03.HDTV.XviD-FUM[ettv].avi" "93B0BDFF98C9E7F7060C0B823D8FE7FB9F9B5226"

Maybe I can do some more digging.

**Comment by mdhiggins** [comment]
Can you post your autoprocess.ini?

**Comment by thejacobsons** [comment]
[uTorrent]
convert = True
couchpotato-label = movies
sickbeard-label = sickbeard
sonarr-label = tv
bypass-label = bypass
sickrage-label = sickrage
webui = True
action_before = stop
action_after = removedata
host = http://localhost:8080/
username = myuid
password = mypwd

**Comment by mdhiggins** [comment]
Can you change your logging level to DEBUG (under the logging.ini file, just change anywhere it says level to DEBUG) and rerun the script and post the uTorrentPostProcess stuff from info.log

**Comment by thejacobsons** [comment]
Ok, I downloaded the latest mp4_automator files. I had errors other than the label error with the new uTorrentPostProcess.py file so I replaced the new one with the old one. It is working from command line now but still isn't working directly from the Run Program setting in uTorrent. The log says it is executing but it never actually does anything unless I run it from the terminal. My next test is to see if I have to put python in front of the command in Run Program.
